pkg/kube/kube_handler.go

```diff
@@ -380,14 +387,26 @@ func getValidationCliAndReq(appRepoBody io.ReadCloser, requestNamespace, kubeapp
 	return cli, req, nil
 }
 
-func (a *userHandler) ValidateAppRepository(appRepoBody io.ReadCloser, requestNamespace string) (*http.Response, error) {
+func doValidationRequest(cli HTTPClient, req *http.Request) (ValidationResponse, error) {
+	res, err := cli.Do(req)
+	if err != nil {
+		// If the request fail, it's not an internal error
+		return ValidationResponse{Code: 400, Message: err.Error()}, nil
+	}
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return ValidationResponse{}, fmt.Errorf("Unable to parse validation response. Got: %v", err)
```

Comment:
If you use a pointer type for the returned `*ValidationResponse` this can just be nil? (and below)